src/main/java/SoftDeleteSample.java

```diff
@@ -200,38 +229,39 @@ public static void deletedSecretRecovery() throws Exception {
         String secretToRecoverName = getRandomName("secret");
         String secretToPurgeName = getRandomName("secret");
 
-        SecretBundle secret = keyVaultClient.setSecret(vault.vaultUri(), secretToRecoverName, "secret to recover");
-        System.out.printf("Create secret %s.%n", secret.toString());
+        SecretAsyncClient secretAsyncClient = vault.secretClient();

```

Comment:
Similarly to above, I would suggest using a sync client.
```suggestion
        SecretClient secretClient = new SecretClientBuilder()
            .credential(createToken())
            .vaultUrl(vault.vaultUri())
            .buildClient();
```